Kernel/Arch/riscv64/CSR.h

```diff
@@ -43,26 +44,30 @@ ALWAYS_INLINE FlatPtr read(Address address)
     return ret;
 }
 
-ALWAYS_INLINE void write(Address address, FlatPtr value)
+template<Address address>
+ALWAYS_INLINE void write(FlatPtr value)
 {
     asm volatile("csrw %0, %1" ::"i"(address), "Kr"(value));
 }
 
-ALWAYS_INLINE FlatPtr read_and_set_bits(Address address, FlatPtr bit_mask)
+template<Address address>
+ALWAYS_INLINE FlatPtr read_and_set_bits(FlatPtr bit_mask)
 {
     FlatPtr ret;
     asm volatile("csrrs %0, %1, %2"
                  : "=r"(ret)
                  : "i"(address), "Kr"(bit_mask));
     return ret;
 }
-
-ALWAYS_INLINE void set_bits(Address address, FlatPtr bit_mask)
+template<Address address, Enum E>
+requires(IsSame<UnderlyingType<E>, u64>)
+ALWAYS_INLINE void set_bits(E bit_mask)
```

Comment:
We might want to use these functions without enums in the future, but I guess we can add an overload for that then.

Reply:
My Idea would be a new `SizedLike<T>` concept, which would abstract that away
Maybe even enforce some kind of triviality